rootfs-builder/rootfs.sh

```diff
@@ -147,7 +147,7 @@ copy_kernel_modules()
 
 OSBUILDER_VERSION="unknown"
 
-while getopts c:ho:r: opt
+while getopts a:ho:r: opt
```

Comment:
Good catch - can you also fetch origin and checkout to that version if is not empty when the agent is cloned

Reply:
Hi @Pennyzct - I think @jcvenegas is asking if you could `git checkout ${AGENT_VERSION}"` since the `-c`/`-a` option wasn't actually doing anything :)

Something like the following:

```diff
$ git diff
diff --git a/rootfs-builder/rootfs.sh b/rootfs-builder/rootfs.sh
index f29d220..a26d504 100755
--- a/rootfs-builder/rootfs.sh
+++ b/rootfs-builder/rootfs.sh
@@ -239,6 +239,7 @@ build_rootfs ${ROOTFS_DIR}
 
 info "Pull Agent source code"
 go get -d "${GO_AGENT_PKG}" || true
+[ -n "${AGENT_VERSION}" ] && git checkout "${AGENT_VERSION}" || true
 OK "Pull Agent source code"
 
 info "Build agent"
``` 

However, this could be handled on a separate PR if you wish.

Reply:
@Pennyzct @jodh-intel ,  yep that was what I had in mind. But yes, could be done in a separate PR.